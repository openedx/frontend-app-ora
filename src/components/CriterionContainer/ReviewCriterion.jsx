import React from 'react';
import PropTypes from 'prop-types';

import { Form, FormControlFeedback } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import messages from './messages';

/**
 * <ReviewCriterion />
 */
const ReviewCriterion = ({ criterion }) => (
  <Form.Group>
    <div className="review-criterion">
      {criterion.options.map((option) => (
        <div className="d-flex text-nowrap">
          <Form.Label className="option-label">{option.name}</Form.Label>
          <div key={option.name} className="criteria-option">
            <div>
              <FormControlFeedback className="option-points">
                <FormattedMessage
                  {...messages.optionPoints}
                  values={{ points: option.points }}
                />
              </FormControlFeedback>
            </div>
          </div>
        </div>
      ))}
    </div>
  </Form.Group>
);

ReviewCriterion.propTypes = {
  criterion: PropTypes.shape({
    options: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      point: PropTypes.number,
    })),
  }).isRequired,
};

export default ReviewCriterion;
