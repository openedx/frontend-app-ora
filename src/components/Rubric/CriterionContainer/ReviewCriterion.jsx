import React from 'react';
import PropTypes from 'prop-types';

import { Form, FormControlFeedback } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import messages from './messages';

/**
 * <ReviewCriterion />
 */
const ReviewCriterion = ({ criterion }) => (
  <div className="review-criterion">
    {criterion.options.map((option) => (
      <div key={option.name} className="criteria-option">
        <div>
          <Form.Label className="option-label">{option.name}</Form.Label>
          <FormControlFeedback className="option-points">
            <FormattedMessage {...messages.optionPoints} values={{ points: option.points }} />
          </FormControlFeedback>
        </div>
      </div>
    ))}
  </div>
);

ReviewCriterion.propTypes = {
  criterion: PropTypes.shape({
    options: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        points: PropTypes.number.isRequired,
      }),
    ).isRequired,
  }).isRequired,
};

export default ReviewCriterion;
