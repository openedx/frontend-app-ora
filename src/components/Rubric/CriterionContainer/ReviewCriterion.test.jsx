import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import ReviewCriterion from './ReviewCriterion';

describe('<ReviewCriterion />', () => {
  const props = {
    criterion: {
      options: [
        {
          name: 'option-1',
          description: 'description-1',
          points: 1,
        },
        {
          name: 'option-2',
          description: 'description-2',
          points: 2,
        },
      ],
    },
  };

  test('renders', () => {
    const wrapper = shallow(<ReviewCriterion {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
    expect(wrapper.instance.findByType('FormControlFeedback').length).toEqual(props.criterion.options.length);
  });
});
