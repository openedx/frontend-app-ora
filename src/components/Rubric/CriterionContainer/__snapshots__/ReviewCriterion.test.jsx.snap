// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCriterion /> renders 1`] = `
<div
  className="review-criterion"
>
  <div
    className="criteria-option"
    key="option-1"
  >
    <div>
      <Form.Label
        className="option-label"
      >
        option-1
      </Form.Label>
      <FormControlFeedback
        className="option-points"
      >
        <FormattedMessage
          defaultMessage="{points} points"
          description="criterion option point value display"
          id="ora-grading.RadioCriterion.optionPoints"
          values={
            Object {
              "points": 1,
            }
          }
        />
      </FormControlFeedback>
    </div>
  </div>
  <div
    className="criteria-option"
    key="option-2"
  >
    <div>
      <Form.Label
        className="option-label"
      >
        option-2
      </Form.Label>
      <FormControlFeedback
        className="option-points"
      >
        <FormattedMessage
          defaultMessage="{points} points"
          description="criterion option point value display"
          id="ora-grading.RadioCriterion.optionPoints"
          values={
            Object {
              "points": 2,
            }
          }
        />
      </FormControlFeedback>
    </div>
  </div>
</div>
`;
